neural_compressor/adaptor/ox_utils/weight_only.py

```diff
@@ -0,0 +1,648 @@
+
+    return model, output_dicts
+
+def apply_awq_clip(model, tune_cfg, absorb_pairs, output_dicts):
+    """Apply clip for weight by checking mse."""
+    ratios = {}
+    for parent, nodes in absorb_pairs.items():
+        if any([node.input[0] not in output_dicts for node in nodes]):
+            logger.warning("Miss tensors of node {} during AWQ, skip it!".format(node.name))
+            continue
+
+        inp = np.concatenate(output_dicts[nodes[0].input[0]], axis=0)
+
+        for node in nodes:
+            config = tune_cfg.get(node.name, {})
+            org_weight = numpy_helper.to_array(
+                                model.get_initializer(node.input[1]),
+                                base_dir=os.path.dirname(model.model_path))
+            org_w_shape = org_weight.shape # ic, oc
+            group_size = config.get("group_size", -1) if config.get("group_size", -1) != -1 else org_w_shape[0]
+            org_out = np.matmul(inp, org_weight) # n_token, oc
+
+            best_error = float("inf")
+            best_ratio = 1
+            for i_s in range(10):
+                ratio = 1 - i_s / 100
+                weight = copy.deepcopy(org_weight)
+                if org_w_shape[0] % group_size == 0:
+                    weight = weight.reshape(group_size, -1)
+                    weight = qdq_tensor(weight, config, ratios.get(node.input[1], 1))
+                    weight = weight.reshape(org_w_shape)
+                else:
+                    index = org_w_shape[0] // group_size * group_size
+                    if index != 0:
+                        part_weight = weight[:index, :].reshape(group_size, -1)
+                        part_weight = qdq_tensor(part_weight, config, ratios.get(node.input[1], 1))
+                        weight[:index, :] = part_weight.reshape(index, -1)
+                    weight[index:, :] = qdq_tensor(weight[index:, :], config, ratios.get(node.input[1], 1))
+                    
+                cur_out = np.matmul(inp, weight)
+                loss = np.mean(np.power((org_out - cur_out), 2))
+                is_best = loss < best_error
+                if is_best:
+                    best_error = loss
+                    best_ratio = ratio
+            ratios[node.input[1]] = best_ratio
+    model = rtn_quantize(model, tune_cfg, ratios)
+    return model
+
+def prepare_inputs(model, n_samples, dataloader):
+    """Prepare inputs for weight only quantization.
+
+    Args:
+        model (ModelProto or ONNXModel): onnx model
+        n_samples (int, optional): calibration sample number.
+        dataloader (object): dataloader for calibration.
+
+    Returns:
+        inputs: prepared inputs.
+        so: session options
+    """
+    from importlib.util import find_spec
+    from neural_compressor.adaptor.ox_utils.util import to_numpy
+    
+    so = ort.SessionOptions()
+    if sys.version_info < (3, 10) and find_spec('onnxruntime_extensions'):  # pragma: no cover

```

Comment:
Just a remind here, `onnxruntime_extensions` is avaliable for python 3.10. I will fix it in #1138 after this PR is merged,